### DISCUSSION TITLE
How to perform a colorspace transformation?

### DISCUSSION THREAD

**Discussion by luoww1992** [colour-science/colour#959] [General]
### Question

i am a newbie, i am learning how to change colorspace from A to B,
now there is a question:
 when i learn to change a log mode photo to sRGB，i know the conversion path is   sd_to_XYZ -->XYZ_to_sRGB.
i am also seeing the  documentation.
now i try change a photo from mode Log3G10 to sRGB
but i don't know to make a photo sd object. maybe it is about log_decoding_Log3G10_v3 or other
so can someone support a demo to show me how to change mode from log3G10 to sRGB by a log mode photo.
thank you very much !

**Comment by nick-shaw** [comment]
You can convert an image from Log3G10 (presumably also RED Wide Gamut) to sRGB like this:
```
>>> from colour.models import (RGB_COLOURSPACE_RED_WIDE_GAMUT_RGB,
...                            RGB_COLOURSPACE_sRGB,
...                            RGB_to_RGB)
>>> from colour import read_image, write_image
>>> log_image = read_image('/path/to/source.tif')
>>> sRGB_image = RGB_to_RGB(log_image,
...                         RGB_COLOURSPACE_RED_WIDE_GAMUT_RGB,
...                         RGB_COLOURSPACE_sRGB,
...                         apply_cctf_decoding=True,
...                         apply_cctf_encoding=True)
>>> write_image(sRGB_image, '/path/to/dest.jpg')
```
However, this is a purely colorimetric transform with no tone mapping. This means that diffuse white in the Log3G10 image will be mapped to peak white in the sRGB image, and anything above that will be clipped. This is not normally what you want.

More usual would be to use a display transform such as one of the RED LUTs or an ACES transform. You can use Colour to apply a LUT, and OCIO can apply ACES transforms.

**Comment by luoww1992** [comment]
yes，you are right !  i have tested some log photos, some photos are wrong, 
i will see what you said how to add lut by colour and OCIO.
thank you !

**Comment by luoww1992** [comment]
@nick-shaw 
i have done it by default cube file in colour :
like this:
    path_name = os.path.splitext(logfile)[0]
    new_name = path_name+'_cube.tif'
    log_image = read_image(logfile)
    if lutfile is None:
        lutfile = 'cube/iridas_cube/ACES_Proxy_10_to_ACES.cube'
    lut = read_LUT(lutfile)
    sRGB_image=lut.apply(log_image)
    write_image(sRGB_image, new_name)

but i have no idea to do by ACES transform. i know very little about it,
can you give me some advices?
